src/daft-core/src/datatypes/mod.rs

```diff
@@ -362,3 +362,11 @@ impl<T: DaftNumericType> DataArray<T> {
         self.as_arrow().values().as_slice()
     }
 }
+
+impl<P: AsRef<str>> FromIterator<Option<P>> for Utf8Array {
```

Comment:
added this to prevent leaking internal arrow2 array details. Previously we were converting to arrow2::Utf8Array, then calling `Utf8Array::from`. Now we can just collect the iterator without the extra step.